internal/testmodels/subschema_switch_without_order.go

```diff
@@ -0,0 +1,33 @@
+package testmodels
+
+import (
+	"github.com/discovery-digital/jsonschema"
+)
+
+// These are models used for the subschema_switch test, but the actual test cases are in reflect_test.go
+type ExampleCaseWithoutOrder struct {
+	Type string `json:"type" jsonschema:"optional"`
+}
+
+type IntPayloadWithoutOrder struct {
+	Payload int `json:"payload"`
+}
+
+type StringPayloadWithoutOrder struct {
+	Payload string `json:"payload"`
+}
+
+type BoolPayloadWithoutOrder struct {
+	Payload bool `json:"payload"`
+}
+
+func (ex ExampleCaseWithoutOrder) Case() jsonschema.SchemaSwitch {
+	cases := make(map[string]interface{})
+	cases["bool"] = BoolPayload{}
+	cases["int"] = IntPayload{}
+	cases["string"] = StringPayload{}
```

Comment:
Let me ask you this, what happens to your code in `subschema_switch.go` when users who use this library do not have the specific field names `bool`, `int`, and `string` returned in their `SchemaSwitch.Cases`?

Reply:
If the order is not specified, if the keys present in cases is also present the orderlist which specifies, the order for reflection. Here can write a mechanism to throw the error, saying the specific keys to be present, else we can create a wrapper around the map for cases (struct) which stores the order in which keys are added.

Reply:
Let's say I am a potential user of this library and would like to leverage this switch functionality so I can validate my **existing** json input against different schema definitions with a jsonschema validator.

I have a non-RESTful API which serves a single-endpoint. A few examples of jsoninputs it handles:
```
{
  "action": "add_user",
  "payload": {
    "first_name": "John",
    "last_name": "Doe",
    "age": 22
  }
}
```
```
{
  "action": "add_users",
  "payload": [
    {
      "age": 22,
      "first_name": "John",
      "last_name": "Doe"
    },
    {
      "age": 32,
      "first_name": "Jane",
      "last_name": "Doe"
    }
  ]
}
```
```
{
  "action": "get_user_list",
  "payload": {
    "format": "csv",
    "gzip": true,
  }
}
```

I recognize that I'm able to map the value of `action` to different schemas that validate against the value of `payload` and wish to use this library. 

I create `jsonschema.SchemaSwitch` with `ByField` set to `action`, add my cases `add_user`, `add_users`, and `get_user_list`, and pass the structs I have mapped for them.

Let's say this library uses your **proposed** solution and then it blows up because none of my actions are the values `int`, `bool`, `string`. Does this make any sense to you?